metadata-ingestion/src/datahub/ingestion/source/sql/clickhouse.py

```diff
@@ -407,54 +465,252 @@ def create(cls, config_dict, ctx):
         config = ClickHouseConfig.parse_obj(config_dict)
         return cls(config, ctx)
 
-    def get_workunits_internal(self) -> Iterable[Union[MetadataWorkUnit, SqlWorkUnit]]:
-        for wu in super().get_workunits_internal():
-            if (
-                self.config.include_table_lineage
-                and isinstance(wu, SqlWorkUnit)
-                and isinstance(wu.metadata, MetadataChangeEvent)
-                and isinstance(wu.metadata.proposedSnapshot, DatasetSnapshot)
-            ):
-                dataset_snapshot: DatasetSnapshotClass = wu.metadata.proposedSnapshot
-                assert dataset_snapshot
-
-                lineage_mcp, lineage_properties_aspect = self.get_lineage_mcp(
```

Comment:
seems like the original code was complex for no reason. I'm going to simplify that here https://github.com/datahub-project/datahub/pull/12442, unless you'd like to integrate that into this PR